trainer_image/Dockerfile

```diff
@@ -0,0 +1,115 @@
+
+ENV CUDA_VERSION=$CUDA_VERSION \
+    NV_CUDA_LIB_VERSION=11.8.0-1 \
+    NVIDIA_VISIBLE_DEVICES=all \
+    NVIDIA_DRIVER_CAPABILITIES=compute,utility \
+    NV_CUDA_CUDART_VERSION=11.8.89-1 \
+    NV_CUDA_COMPAT_VERSION=520.61.05-1
+
+RUN dnf config-manager \
+       --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo \
+    && dnf install -y \
+        cuda-cudart-11-8-${NV_CUDA_CUDART_VERSION} \
+        cuda-compat-11-8-${NV_CUDA_COMPAT_VERSION} \
+    && echo "/usr/local/nvidia/lib" >> /etc/ld.so.conf.d/nvidia.conf \
+    && echo "/usr/local/nvidia/lib64" >> /etc/ld.so.conf.d/nvidia.conf \
+    && dnf clean all
+
+ENV CUDA_HOME="/usr/local/cuda" \
+    PATH="/usr/local/nvidia/bin:${CUDA_HOME}/bin:${PATH}" \
+    LD_LIBRARY_PATH="/usr/local/nvidia/lib:/usr/local/nvidia/lib64:$CUDA_HOME/lib64:$CUDA_HOME/extras/CUPTI/lib64:${LD_LIBRARY_PATH}"
+
+
+ENV NV_NVTX_VERSION=11.8.86-1 \
+    NV_LIBNPP_VERSION=11.8.0.86-1 \
+    NV_LIBCUBLAS_VERSION=11.11.3.6-1 \
+    NV_LIBNCCL_PACKAGE_VERSION=2.15.5-1+cuda11.8
+
+RUN dnf config-manager \
+       --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo \
+    && dnf install -y \
+        cuda-libraries-11-8-${NV_CUDA_LIB_VERSION} \
+        cuda-nvtx-11-8-${NV_NVTX_VERSION} \
+        libnpp-11-8-${NV_LIBNPP_VERSION} \
+        libcublas-11-8-${NV_LIBCUBLAS_VERSION} \
+        libnccl-${NV_LIBNCCL_PACKAGE_VERSION} \
+    && dnf clean all
+
+ENV NV_CUDA_CUDART_DEV_VERSION=11.8.89-1 \
+    NV_NVML_DEV_VERSION=11.8.86-1 \
+    NV_LIBCUBLAS_DEV_VERSION=11.11.3.6-1 \
+    NV_LIBNPP_DEV_VERSION=11.8.0.86-1 \
+    NV_LIBNCCL_DEV_PACKAGE_VERSION=2.15.5-1+cuda11.8
+
+RUN dnf config-manager \
+       --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo \
+    && dnf install -y \
+        cuda-command-line-tools-11-8-${NV_CUDA_LIB_VERSION} \
+        cuda-libraries-devel-11-8-${NV_CUDA_LIB_VERSION} \
+        cuda-minimal-build-11-8-${NV_CUDA_LIB_VERSION} \
+        cuda-cudart-devel-11-8-${NV_CUDA_CUDART_DEV_VERSION} \
+        cuda-nvml-devel-11-8-${NV_NVML_DEV_VERSION} \
+        libcublas-devel-11-8-${NV_LIBCUBLAS_DEV_VERSION} \
+        libnpp-devel-11-8-${NV_LIBNPP_DEV_VERSION} \
+        libnccl-devel-${NV_LIBNCCL_DEV_PACKAGE_VERSION} \
+    && dnf clean all
+
+ENV LIBRARY_PATH="$CUDA_HOME/lib64/stubs"
+
+RUN dnf install -y python3.11 git && \
+    ln -s /usr/bin/python3.11 /bin/python && \
+    python -m ensurepip --upgrade
+
+RUN mkdir /app
+
+WORKDIR /tmp
+RUN python -m pip install packaging && \
+    python -m pip install --upgrade pip && \
+    python -m pip install torch && \
+    python -m pip install wheel
+RUN git clone https://github.com/foundation-model-stack/fms-hf-tuning.git && \
+    cd fms-hf-tuning && \
+    python -m pip install -r requirements.txt && \
+    python -m pip install -U datasets  && \
+    python -m pip install /tmp/fms-hf-tuning
```

Comment:
Instead of cloning the repo its in, it should COPY in the files needed and install directly. It shouldn't need the `tuning` dir since that is pip installed although having this is useful for testing. But including the `scripts/run_inference` would be useful

Reply:
That's fine with me, will you add a TODO note in the Dockerfile